src/main/java/com/direwolf20/mininggadgets/common/items/MiningGadget.java

```diff
@@ -338,6 +344,10 @@ public void onUsingTick(ItemStack stack, LivingEntity player, int count) {
         Level world = player.level;
         if (world.isClientSide) {
             this.playLoopSound(player, stack);
+        } else {
+            if (UpgradeTools.containsActiveUpgradeFromList(UpgradeTools.getUpgrades(stack), Upgrade.BATTERY_CREATIVE)) {
+                stack.getCapability(ForgeCapabilities.ENERGY).ifPresent(e -> e.receiveEnergy(e.getMaxEnergyStored() - e.getEnergyStored(), false));
```

Comment:
This would cause the NBT of the item to be updated each tick? I think a better solution would be to add a flag to the cap and make it so the cap will bypass actually using energy when the creative upgrade is present. This would go along with evaulating the performance of the contains upgrade performance. 

It's possible it's simpler to add a root flag to the NBT to hold if the gadget is in creative mode or not. We control the only way of adding upgrades so it's relatively simple to hook the logic for when the upgrade is added and removed.

Reply:
Since the laser uses no energy per pick I've removed this code, it was just nice to have it always showing max energy